javascript/code-challenges/challenges-01.test.js

```diff
@@ -0,0 +1,214 @@
+  arr.forEach((string) => localArray.push(upperCase(string)));
+
+  return localArray;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named `greeting` that takes in a single string and returns the string in all uppercase letters, and followed by an "!".
+
+Then, write a function named `speaker` that takes in an array of strings and a callback function. 
+
+Use `forEach` to build a new array of strings, each string modified by the callback. Return the new array. 
+------------------------------------------------------------------------------------------------ */
+
+const greeting = (word) => {
+  return word.toUpperCase() + '!';
+};
+
+const speaker = (words, callback) => {
+  const newArray = [];
+  words.forEach(word => newArray.push(callback(word)));
+  return newArray;
+  
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named addValues that takes in an array and a value and pushes the value into the array. This function does not need a return statement.
+
+Then, write a function named addNumbers that takes in four arguments:
+  - A number to be added to an array
+  - An array into which the number should be added
+  - The number of times the number should be added
+  - A callback function to use to add the numbers to the array (Hint: you already defined it)
+
+Within the addNumbers function, invoke the callback function as many times as necessary, based on the third argument of the addNumbers function.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (array, value) => {
+  array.push(value);
+};
+
+const addNumbers = (num, array1, times, callback) => {
+  
+  for (let i = 0; i < times; i += 1){
+    callback(array1, num);
+  }
+   
+  
+  return array1;
+};
+
+/* ------------------------------------------------------------------------------------------------
+
+CHALLENGE 6
+
+Write a function named createList that takes in an array of the current store inventory.
+
+The inventory is formatted like this:
+[
+  { name: 'apples', available: true },
+  { name: 'pears', available: true },
+  { name: 'oranges', available: false },
+  { name: 'bananas', available: true },
+  { name: 'blueberries', available: false }
+]
+
+This function should use forEach to populate your grocery list based on the store's inventory. If the item is available, add it to your list. Return the final list.
+------------------------------------------------------------------------------------------------ */
+// With help from Spencer Tower and Erik Savage
+const createList = (availableItems) => {
+  //const test = available => (available === true);
+  const newArray =[];
+  availableItems.forEach(fruitObject => {  if (fruitObject.available === true ) {newArray.push(fruitObject.name) } })
```

Comment:
Really nice work on all the challenges!

Feedback here would be to indent the if statement on line 129 to follow JS convention/best practices. Curlies aren't needed for one line statements but in this case, since we have one, we can indent so this is a bit easier to read (in general indenting is good between curlies)👍